spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/thymeleaf/ThymeleafAutoConfiguration.java

```diff
@@ -159,18 +152,6 @@ public LayoutDialect layoutDialect() {
 	}
 
 	@Configuration
-	@ConditionalOnClass(DataAttributeDialect.class)
```

Comment:
@zhanhb AFAIK, this is gonna by handled by `thymeleaf` itself. I am waiting to test everything and then know if it will be removed or not.